crates/types/src/image.rs

```diff
@@ -1,33 +1,65 @@
-use color_eyre::Result;
-use serde::{Deserialize, Serialize};
+use color_eyre::{eyre::Context, Result};
+use serde::{
+    de::{self, Visitor},
+    ser::SerializeStruct,
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+use serde_bytes::{ByteBuf, Bytes};
 use serialize_hierarchy::SerializeHierarchy;
 use std::{
-    fmt::{Debug, Error, Formatter},
+    fmt::{self, Debug, Formatter},
     mem::{size_of, ManuallyDrop},
     ops::Index,
     path::Path,
     sync::Arc,
 };
 
-use image::{io::Reader, RgbImage};
+use image::{
+    codecs::jpeg::JpegEncoder, io::Reader, load_from_memory_with_format, ImageFormat, RgbImage,
+};
 use nalgebra::Point2;
 
 use crate::{Rgb, YCbCr444};
 
 use super::color::YCbCr422;
 
-#[derive(Clone, Default, Deserialize, Serialize, SerializeHierarchy)]
+const SERIALIZATION_JPEG_QUALITY: u8 = 40;
+
+#[derive(Clone, Default, SerializeHierarchy)]
 pub struct Image {
     buffer: Arc<Vec<YCbCr422>>,
     width_422: u32,
     height: u32,
 }
 
 impl Debug for Image {
-    fn fmt(&self, formatter: &mut Formatter<'_>) -> Result<(), Error> {
+    fn fmt(&self, formatter: &mut Formatter<'_>) -> Result<(), fmt::Error> {
+        struct DebugBuffer {
+            buffer_length: usize,
+        }
+
+        impl Debug for DebugBuffer {
+            fn fmt(&self, formatter: &mut Formatter<'_>) -> Result<(), fmt::Error> {
+                formatter.write_fmt(format_args!(
+                    "[{} element{}]",
```

Comment:
Perhaps be more specific?
```suggestion
                    "[{} pixel{}]",
```

Reply:
Done